client/index.js

```diff
@@ -17,43 +17,58 @@ var sock = null;
 var hot = false;
 var initial = true;
 var currentHash = "";
+var quiet = false;
+var noInfo = true;
+var logLevel = 3; // 3 = all, 2 = warnings and errors, 1 = only errors, 0 = quiet
+
+function log(level, msg) {
+	if(logLevel >= 3 && level === "info")
+		return console.log(msg);
+	if(logLevel >= 2 && level === "warning")
+		return console.warn(msg);
+	if(logLevel >= 1 && level === "error")
+		return console.error(msg);
+}
 
 var onSocketMsg = {
 	hot: function() {
 		hot = true;
-		console.log("[WDS] Hot Module Replacement enabled.");
+		log("info", "[WDS] Hot Module Replacement enabled.");
 	},
 	invalid: function() {
-		console.log("[WDS] App updated. Recompiling...");
+		log("info", "[WDS] App updated. Recompiling...");
 	},
 	hash: function(hash) {
 		currentHash = hash;
 	},
 	"still-ok": function() {
-		console.log("[WDS] Nothing changed.")
+		log("info", "[WDS] Nothing changed.")
+	},
+	"log-level": function(level) {
+		logLevel = level;
 	},
 	ok: function() {
 		if(initial) return initial = false;
 		reloadApp();
 	},
 	warnings: function(warnings) {
-		console.log("[WDS] Warnings while compiling.");
+		log("info", "[WDS] Warnings while compiling.");
 		for(var i = 0; i < warnings.length; i++)
-			console.warn(stripAnsi(warnings[i]));
+			log("warn", stripAnsi(warnings[i]));
```

Comment:
These are useful warnings and I want them to be reported. They are related to user’s code.